assets/js/types/type-defs/contexts.ts

```diff
@@ -13,6 +13,12 @@ export type ValidationData = {
 	) => void;
 };
 
+export type ProductQueryContext = {
+	query: object;
```

Comment:
In general, `object` should never be used as a type. TSLint used to report it by default, and I just noticed [it doesn't anymore since last year](https://github.com/woocommerce/woocommerce-blocks/pull/7366).

In any case, `object` indicates also arrays, for instance. In general, you can use the utility type `Record` for your object needs.

In this case, we do have the `query` type: you can use `QueryBlockQuery & ProductQueryArguments`.

Reply:
Addressed in ad2b4decd.

Thanks, this was very helpful! 🙂